pkg/cloudprovider/azure.go

```diff
@@ -339,29 +350,30 @@ func (a *Azure) GetNodeEgressIPConfiguration(node *corev1.Node, cloudPrivateIPCo
 	return []*NodeEgressIPConfiguration{config}, nil
 }
 
-func (a *Azure) createOrUpdate(networkInterface network.Interface) (network.InterfacesCreateOrUpdateFuture, error) {
+func (a *Azure) createOrUpdate(networkInterface armnetwork.Interface) error {
 	ctx, cancel := context.WithTimeout(a.ctx, defaultAzureOperationTimeout)
 	defer cancel()
-	return a.networkClient.CreateOrUpdate(ctx, a.resourceGroup, *networkInterface.Name, networkInterface)
-}
+	_, err := a.networkClient.BeginCreateOrUpdate(ctx, a.resourceGroup, ptr.Deref(networkInterface.Name, ""), networkInterface, nil)
```

Comment:
so you removed the `a.waitForCompletion(result)` from all of the calls to `a.waitForCompletion(result)`.
This means to me that we are no longer waiting for the operation to complete, why?
`BeginCreateOrUpdate` returns a poller that could help achieve the same behavior as we had and I think we should use it here.